ui/reactAngularAdaptor.js

```diff
@@ -1,24 +1,53 @@
 import {react2angular} from 'react2angular';
 import AppointmentContainer from './react-components/containers/AppointmentContainer.jsx';
+import CancelConfirmationWrapper from "./react-components/components/CancelConfirmation/CancelConfirmationWrapper.jsx";
 
 angular.module('bahmni.appointments')
 .component('reactAddAppointment', react2angular(AppointmentContainer));
 
+angular.module('bahmni.appointments')
+    .component('cancelConfirmationWrapper', react2angular(CancelConfirmationWrapper));
+
 // Used this component to pass a callback function to react component
 angular.module('bahmni.appointments').component('reactAddAppointmentWrapper',{
-    template: '<react-add-appointment on-back="onBack" set-view-date="setViewDate" appointment-uuid="appointmentUuid" is-recurring="isRecurring">>',
+    template: '<react-add-appointment on-back="onBack" set-view-date="setViewDate" appointment-uuid="appointmentUuid" is-recurring="isRecurring" state="state">',
     controller: reactAddAppointmentController
 });
 
-reactAddAppointmentController.$inject = ['$scope', '$state', '$stateParams'];
-function reactAddAppointmentController($scope, $state, $stateParams) {
+angular.module('bahmni.appointments').run(['$templateCache', function ($templateCache) {
+    $templateCache.put('templateId', '<cancel-confirmation-wrapper close="onClose" on-back="onBack">');
+}]);
+
+reactAddAppointmentController.$inject = ['$rootScope', '$location', '$scope', '$state', 'ngDialog', '$stateParams'];
+function reactAddAppointmentController($rootScope, $location, $scope, $state, ngDialog, $stateParams) {
+    let onBack = false;
+    let backUrl = "^";
     $scope.onBack = function () {
-        $state.go('^', $state.params, {reload: true});
+        onBack = true;
+        $state.go(backUrl, $state.params, {reload: true});
+        ngDialog.close();
     };
 
+    $state.showCancelConfirmation = false;
+    $scope.$on('$stateChangeStart', function (event, next, current) {
+        if (next.url !== "/new" && !onBack) {
```

Comment:
can you check if it works for edit as well?

Reply:
Okay will check

Reply:
Working fine with edit as well. @bsneha90